### ISSUE TITLE
Numerical issue in fastmnmf

### ISSUE THREAD

**Issue by fakufaku** [LCAV/pyroomacoustics#255]
There is a numerical error for the maximum frame length in the tests for FastMNMF (see the log below).

I think it could be fixed by adding an `eps` to the denominator of the expression on [line 170](https://github.com/LCAV/pyroomacoustics/blob/204177d494e6efcc9461e7c86cf2a85f74f3fd33/pyroomacoustics/bss/fastmnmf.py#L170).

@sekiguchi92 Can I go ahead and add the epsilon or can it be a problem for the algorithm ?

```
=================================== FAILURES ===================================
__________________________ TestBSS.test_bss_fastmnmf ___________________________

self = <test_bss.TestBSS testMethod=test_bss_fastmnmf>

    def test_bss_fastmnmf(self):
        for block in L:
>           freq_bss(algo="fastmnmf", L=block)

pyroomacoustics/bss/tests/test_bss.py:168: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

algo = 'fastmnmf', L = 4096, kwargs = {}, room_dim = [8, 9]
room = <pyroomacoustics.room.ShoeBox object at 0x1297c23d0>
signals = [array([-21., -21., -28., ..., 129., 110., 100.], dtype=float32), array([29., 34., 38., ..., 53., 54., 52.], dtype=float32)]
delays = [1.0, 0.0], locations = [[2.5, 3], [2.5, 6]]
sig = array([29., 34., 38., ..., 53., 54., 52.], dtype=float32), d = 0.0

    def freq_bss(algo="auxiva", L=256, **kwargs):
    
        # Room dimensions in meters
        room_dim = [8, 9]
    
        # create a room with sources and mics
        room = pra.ShoeBox(room_dim, fs=16000, max_order=0, sigma2_awgn=1e-8)
    
        # get signals
        signals = [
            np.concatenate([wavfile.read(f)[1].astype(np.float32) for f in source_files])
            for source_files in wav_files
        ]
        delays = [1.0, 0.0]
        if algo == "overiva":
            locations = [[2.5, 3]]
        else:
            locations = [[2.5, 3], [2.5, 6]]
    
        # add mic and good source to room
        # Add silent signals to all sources
        for sig, d, loc in zip(signals, delays, locations):
            room.add_source(loc, signal=np.zeros_like(sig), delay=d)
    
        # add microphone array
        room.add_microphone_array(
            pra.MicrophoneArray(np.c_[[6.5, 4.49], [6.5, 4.51]], fs=room.fs)
        )
    
        # compute RIRs
        room.compute_rir()
    
        # Record each source separately
        separate_recordings = []
        for source, signal in zip(room.sources, signals):
    
            source.signal[:] = signal
    
            room.simulate()
            separate_recordings.append(room.mic_array.signals)
    
            source.signal[:] = 0.0
        separate_recordings = np.array(separate_recordings)
    
        # Mix down the recorded signals
        mics_signals = np.sum(separate_recordings, axis=0)
    
        ## STFT analysis
        # shape == (n_chan, n_frames, n_freq)
        X = pra.transform.stft.analysis(
            mics_signals.T, L, L, zp_front=L // 2, zp_back=L // 2
        )
    
        ## START BSS
        if algo == "auxiva":
            # Run AuxIVA
            Y = pra.bss.auxiva(X, n_iter=30, proj_back=True, **kwargs)
            max_mse = 5e-2
        elif algo == "ilrma":
            # Run ILRMA
            Y = pra.bss.ilrma(X, n_iter=30, n_components=2, proj_back=True, **kwargs)
            max_mse = 5e-2
        elif algo == "sparseauxiva":
            # Estimate set of active frequency bins
            ratio = 0.35
            average = np.abs(np.mean(np.mean(X, axis=2), axis=0))
            k = np.int_(average.shape[0] * ratio)
            S = np.sort(np.argpartition(average, -k)[-k:])
            # Run SparseAuxIva
            Y = pra.bss.sparseauxiva(X, S, n_iter=30, proj_back=True, **kwargs)
            max_mse = 1.5e-1
        elif algo == "overiva":
            Y = pra.bss.auxiva(X, n_src=1, n_iter=30, proj_back=True, **kwargs)
            max_mse = 0.5
        elif algo == "fastmnmf":
            Y = pra.bss.fastmnmf(X, n_iter=30, n_components=2, **kwargs)
            max_mse = 1e-1
        elif algo == "fastmnmf2":
            Y = pra.bss.fastmnmf2(X, n_iter=30, n_components=2, **kwargs)
            max_mse = 1e-1
    
        ## STFT Synthesis
        if algo == "overiva":
            y = pra.transform.stft.synthesis(
                Y[:, :, 0], L, L, zp_front=L // 2, zp_back=L // 2
            ).T
            y = y[None, :]
        else:
            y = pra.transform.stft.synthesis(Y, L, L, zp_front=L // 2, zp_back=L // 2).T
    
        # Calculate MES
        #############
        ref = np.moveaxis(separate_recordings, 1, 2)
        y_aligned = y[:, L // 2 : ref.shape[1] + L // 2]
    
        mse = np.mean((ref[:, : y_aligned.shape[1], 0] - y_aligned) ** 2)
        ref_var = np.var(np.concatenate(ref[:, : y_aligned.shape[1], 0]))
    
        print(
            "%s with a %d frame length: Relative MSE (expected less than %.e)"
            % (algo, L, max_mse),
            mse / ref_var,
        )
>       assert (mse / ref_var) < max_mse
E       assert (nan / 421650.92284736375) < 0.1

pyroomacoustics/bss/tests/test_bss.py:129: AssertionError
----------------------------- Captured stdout call -----------------------------
fastmnmf with a 256 frame length: Relative MSE (expected less than 1e-01) 0.002166374053055612
fastmnmf with a 512 frame length: Relative MSE (expected less than 1e-01) 0.0025593736456267432
fastmnmf with a 1024 frame length: Relative MSE (expected less than 1e-01) 0.0032127825593538713
fastmnmf with a 2048 frame length: Relative MSE (expected less than 1e-01) 0.04244914879554848
fastmnmf with a 4096 frame length: Relative MSE (expected less than 1e-01) nan
=============================== warnings summary ===============================
pyroomacoustics/bss/tests/test_bss.py::TestBSS::test_bss_fastmnmf
  /Users/runner/work/pyroomacoustics/pyroomacoustics/pyroomacoustics/bss/fastmnmf.py:171: RuntimeWarning: divide by zero encountered in true_divide
    tmp_FM
```

**Comment by fakufaku** [comment]
@sekiguchi92 I have confirmed that the test pass when I add the eps in the denominator on line 150. Can you confirm that this okay algorithmically as well ?

**Comment by sekiguchi92** [comment]
I'm sorry for bothering you.
Yes, I think it's OK.
Thank you very much.

**Comment by fakufaku** [comment]
Great! Thanks for the quick answer!
Fixed in #254
